### PR TITLE
Data Generator: performance enhancement, joinable schemas and distribution matching

### PR DESCRIPTION
**Pull request by jaketf**
This Pull Requests adds the following:

A distribution matching pipeline and associated histogram tool for performance testing
A pipeline for generating joinable schemas
Many needed documentation updates
The ability to constrain string length by parsing from the description field of BQ schema (for example if you wish to generate data to load into RDBMS)
This Pull Request renames DataGenerator.py to PrettyDataGenerator.py and introduces a very similar PerformantDataGenerator.py which is mostly the same but removes the usage of the Faker library for faster random data generation. The associated tests scripts have been updated to reflect this change.

I would like two reviewers @pupamanyu who is familiar with the related code changes from #117 and @joshmcginley who I'd like to provide feedback regarding the documentation.

### CODE REVIEW

**Comment by jaketf** [comment]
@ryanmcdowell please add @pupamanyu as a reviewer.

**Comment by jaketf** [comment]
@ryanmcdowell I wasn't able to add @pupamanyu as a reviewer. Pramod has agreed to review this.

**Comment by ryanmcdowell** [comment]
@jaketf I'm unable to add him either. He either has to self-assign or join the GCP org with his GitHub account so he is assignable.

File: examples/dataflow-data-generator/data-generator-pipeline/data_generator/PerformantDataGenerator.py [hunk 1]
```diff
@@ -0,0 +1,753 @@
...
+            element: A single element of the PCollection 
+        """
+
+        faker_schema = self.data_gen.get_faker_schema()
+        try:
+            # Here the element is treated as the dictionary representing a single row
+            # of the histogram table.
+            frequency = element.get(u'frequency')
+
+            for i in xrange(int(frequency)):
```
[the remark this answers is not in the document]
Comment (jaketf) [reply examples/dataflow-data-generator/data-generator-pipeline/data_generator/PerformantDataGenerator.py]: Done.

**Comment by pupamanyu** [comment]
I have reviewed the code. LGTM approved @ryanmcdowell

**Pull request merged**